ovh/resource_cloud_project_containerregistry_ip_restrictions_management.go

```diff
@@ -0,0 +1,175 @@
+package ovh
+
+import (
+	"errors"
+	"fmt"
+	"log"
+	"net/url"
+	"strings"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/jackc/pgtype"
+	"github.com/ovh/terraform-provider-ovh/ovh/helpers"
+)
+
+func resourceCloudProjectContainerRegistryIPRestrictionsManagement() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceCloudProjectContainerRegistryIPRestrictionsManagementPut,
+		Delete: resourceCloudProjectContainerRegistryIPRestrictionsManagementDelete,
+		Update: resourceCloudProjectContainerRegistryIPRestrictionsManagementPut,
+		Read:   resourceCloudProjectContainerIPRestrictionsManagementRead,
+		Importer: &schema.ResourceImporter{
+			State: resourceCloudProjectContainerRegistryIPRestrictionsManagementImportState,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"service_name": {
+				Type:        schema.TypeString,
+				Description: "Service name",
+				ForceNew:    true,
+				Required:    true,
+				DefaultFunc: schema.EnvDefaultFunc("OVH_CLOUD_PROJECT_SERVICE", nil),
+			},
+			"registry_id": {
+				Type:        schema.TypeString,
+				ForceNew:    true,
+				Description: "RegistryID",
+				Required:    true,
+			},
+			"ip_restrictions": {
+				Type:        schema.TypeList,
+				Description: "List your IP restrictions applied on artifact manager component",
+				Required:    true,
+				Elem: &schema.Schema{
+					Type: schema.TypeMap,
+					Set:  schema.HashString,
+					ValidateFunc: func(ipRestrictionInterface interface{}, path string) (warning []string, errorList []error) {
+						ipRestriction := ipRestrictionInterface.(map[string]interface{})
+
+						if ipRestriction["ip_block"] == nil {
+							return nil, []error{errors.New(fmt.Sprintf("ipBlock attribute is mandatory for ip_restrictions: %s", path))}
+						}
+
+						ipBlockString := ipRestriction["ip_block"].(string)
+						var cidr pgtype.Inet
+
+						if err := cidr.Set(ipBlockString); err != nil {
+							return nil, []error{fmt.Errorf("ip_block: %s is not CIDR format", ipBlockString)}
+						}
+
+						return nil, nil
+
+					},
+				},
+			},
+		},
+	}
+}
+
+func resourceCloudProjectContainerRegistryIPRestrictionsManagementImportState(d *schema.ResourceData, meta interface{}) ([]*schema.ResourceData, error) {
+	givenId := d.Id()
+	log.Printf("[DEBUG] Importing cloud project registry IP restrictions of management type %s", givenId)
+
+	splitId := strings.SplitN(givenId, "/", 3)
+
+	if len(splitId) != 2 {
+		return nil, fmt.Errorf("import Id is not service_name/registryid formatted")
+	}
+
+	serviceName := splitId[0]
+	registryID := splitId[1]
+
+	d.SetId(serviceName + "/" + registryID)
```

Comment:
not needed ? From what I understand `d.Id` should already contain exactly this

Reply:
Not really sure, I see it same on different resources.